### PR TITLE
Remove cached_property dependency for python 3.8

### PR DESCRIPTION
**Pull request by wakemaster39**
Currently not using any of the added stuff from cached_property inside of the class. Use the builtin from functools if python 3.8 or newer.

Took first step in punycode by ensuring FQDN is lowered on consturction.

### CODE REVIEW

**Comment by wakemaster39** [comment]
I also made the `__str__` based on absolute and not the `_fqdn` property. This ensures alignment with equality and hashes so you don't call str(SomeFQDN) and get two different values.

File: fqdn/_compat.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import sys
+
+print(sys.version_info[:2])
```
[the remark this answers is not in the document]
Comment (wakemaster39) [reply fqdn/_compat.py:3]: yeah for debugging.

File: fqdn/__init__.py [hunk 2]
```diff
@@ -34,10 +34,10 @@ class FQDN:
     def __init__(self, fqdn):
         if not (fqdn and isinstance(fqdn, str)):
             raise ValueError("fqdn must be str")
-        self._fqdn = fqdn
+        self._fqdn = fqdn.lower()
 
     def __str__(self):
-        return self._fqdn
+        return self.absolute
```
[the remark this answers is not in the document]
Comment (wakemaster39) [reply fqdn/__init__.py:40]: Can return relative instead, my primary use case right now is with DNS/IP management so absolute it far more important to me than relative but I also don't use `str(XXX)` much.

[the remark this answers is not in the document]
Comment (ypcrts) [reply fqdn/__init__.py]: Hmm. This is my use case too. I think it's everyone's use case, really.

**Pull request merged**